src/eyepop/endpoint.ts

```diff
@@ -624,44 +809,78 @@ export class Endpoint {
+                // create a sandbox if needed
+                if (this._options.isSandbox && this._sandboxId === null) {
+                    const createSandboxUrl = `${this._baseUrl}/sandboxes`;
+                    this._requestLogger.debug('before POST %s', createSandboxUrl);
+                    const response = await this._client.fetch(createSandboxUrl, {
+                        method: 'POST',
+                        headers: headers
+                    });
+                    const responseJson = await response.json();
+                    this._requestLogger.debug('after POST %s', createSandboxUrl);
+                    this._sandboxId = responseJson;
                 }
-                this._requestLogger.debug('before GET %s', get_url)
-                let response = await this._client.fetch(get_url, {
+
+                if (this._options.popId == TransientPopId.Transient) {
+                    this._pipelineId = await this.startPopLessPipeline()
+                    this._popName = this._options.popId
+                } else {
+                    this._pipelineId = config.pipeline_id
+                    this._popName = config.name
+                }
+                this._requestLogger.debug('after GET %s: %s / %s', config_url, this._baseUrl, this._pipelineId)
+                if (!this._pipelineId || !this._baseUrl) {
+                    return Promise.reject(`Pop not started`)
+                }
+
+                let start_pipeline_url;
+                if (this._sandboxId === null) {
+                    start_pipeline_url = `${this._baseUrl}/pipelines`;
+                } else {
+                    start_pipeline_url = `${this._baseUrl}/pipelines?sandboxId=${this._sandboxId}`;
+                }
+
+                headers = {
+                    'Authorization': await this.authorizationHeader()
+                }
+                this._requestLogger.debug('before GET %s', start_pipeline_url)
+                let response = await this._client.fetch(start_pipeline_url, {
                     method: 'GET',
                     headers: headers
                 })
                 if (response.status > 200) {
                     const message = await response.text()
                     return Promise.reject(`Unexpected status ${response.status}: ${message}`)
                 }
-                this._requestLogger.debug('after GET %s', get_url)
-                this._pipeline = (await response.json()) as Pipeline
+                this._requestLogger.debug('after GET %s', start_pipeline_url)
+
+                // ////////////////////////////////////////////////////////
+                //
+                //
+                //
+                // NOTE: json()[0] is a hack that should be removed
+                //        
+                //
+                //
+                // ////////////////////////////////////////////////////////
+                this._pipeline = (await response.json())[0] as Pipeline
```

Comment:
Here is the culprit

Reply:
not sure why we need this

Reply:
I was using the wrong pipeline url - fix found here:
https://github.com/eyepop-ai/eyepop-sdk-node/pull/30/commits/e961cd05c30eeb0c986d987e8d471ff8881a6869